### DISCUSSION TITLE
Reduce padding height when only using digits?

### DISCUSSION THREAD

**Discussion by kiddingme1** [Bodmer/TFT_eSPI#2221] [Q&A - Fonts] [answered]
I think padding height is probably the wrong term. I’m using the Orbitron font and printing only digits in certain areas. The digits have a large number of pixels above and below that get wiped out and leave that area of the display unusable. I suspect it is because characters in that font could consume taller amounts of pixels. 

I want to cram data closer together vertically. Is there anything I can do to reduce the vertical spacing between digits displayed without wiping out part of the pixels? 
![D8B1BA6C-396D-4D3F-B109-E23C17FB56B5](https://user-images.githubusercontent.com/67800879/205377200-861d91d6-6d2e-4ed7-bd2d-3abff48bfc7f.jpeg)

**Comment by Bodmer** [comment, marked as answer]
One way is to create a new font with the metrics set appropriately.

The simplest approach in this case is to use the viewport feature of the library. This restricts rendering to a particular rectangular zone. See [example](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/Viewport_Demo). See Viewport_commands tab of that example.

Here is how I would use it to create a tight box around the digits:
```
  int16_t x = 100;
  int16_t y = 20;

  tft.setTextPadding(53);
  tft.setTextDatum(TL_DATUM);
  tft.setViewport(x, y+9, 53, 24, false);
  tft.setFreeFont(&Orbitron_Light_32);
  for( int i = 100; i > 0; i--)
  {
    tft.drawNumber( i, x, y);
    delay(50);
  }
  tft.resetViewport();
  tft.setTextPadding(0);
```
You will need to adjust the viewport  box to be tight around the digits by experimentation or by looking at the [glyph metrics.](https://github.com/Bodmer/TFT_eSPI/blob/master/Fonts/Custom/Orbitron_Light_32.h#L101-L128)

If you use a different text datum the viewport position will need to be adjusted accordingly. In practice I expect you ony need a viewport restriction in the y direction. so in the above case  tft.setViewport(0, y+9, tft.width(), 24, false); would work.

**Reply by kiddingme1** [reply]
Thank you Bodmer!
